subsys/bluetooth/controller/hci/hci.c

```diff
@@ -6548,6 +6549,22 @@ static void le_per_adv_sync_report(struct pdu_data *pdu_data,
 		return;
 	}
 
+	/* NOTE: The timeout_reload field in the sync context is checked under
+	 *       race condition between HCI Tx and Rx thread wherein a sync
+	 *       terminate was performed which resets the timeout_reload field
+	 *       before releasing the sync context back into its memory pool.
```

Comment:
Should it be `volatile` then?

Reply:
It is `volatile`, https://github.com/zephyrproject-rtos/zephyr/blob/40d6fe0ef62403c8a2396f34a38ed191652852a3/subsys/bluetooth/controller/ll_sw/ull_sync_types.h#L21

Reply:
Ah, so if the field is volatile, then the pointer to the struct doesn't need to be? That's good to know :)

Reply:
The structure's indirect addressing does not change, only the value at the member offset (here `timeout_reload`) is `volatile`.